csrc/host_ir/pass/insert_deallocations.cpp

```diff
@@ -23,6 +23,18 @@ void InsertDeallocations::passImplementation(Fusion* fusion) {
         "anyways");
   });
   std::unordered_map<TensorView*, int64_t> last_use;
+  std::unordered_set<TensorView*> non_intermediate_tensors;
+  for (auto* val : hic->inputs()) {
+    if (auto* tv = val->as<TensorView>()) {
+      non_intermediate_tensors.insert(tv);
+    }
+  }
+  for (auto* val : hic->outputs()) {
+    if (auto* tv = val->as<TensorView>()) {
+      non_intermediate_tensors.insert(tv);
+    }
+  }
+
   for (auto&& [i, expr] : enumerate(top_level_exprs)) {
     for (auto* val : expr->inputs()) {
```

Comment:
```suggestion
    for (auto* tv : ir_utils::filterByType<TensorView>(expr->inputs())) {
```
so we can remove the call to `isA` and `as`.